lib/chef/resource/alternatives.rb

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright:: 2020, Chef Software Inc.
+# Copyright:: 2016-2020, Virender Khatri
+#
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../resource"
+
+class Chef
+  class Resource
+    class Alternatives < Chef::Resource
+      unified_mode true
+
+      provides(:alternatives) { true }
+
+      description "The alternatives resource allows for configuration of command alternatives in Linux using the alternatives or update-alternatives packages."
+      introduced "16.0"
+
+      property :link_name, String, name_property: true
+      property :link, String, default: lazy { |n| "/usr/bin/#{n.link_name}" }
+      property :path, String
+      property :priority, [String, Integer], coerce: proc { |n| n.to_i }
```

Comment:
If this is getting coerced with `to_i`, can it ever be a String?

Reply:
The user can give it to us as a string and then we just make it an Integer immediately.

Reply:
I must have been reading the docs wrong, I thought it happened in the opposite order.